tests/fixtures/test-parse.css

```diff
@@ -43,8 +43,8 @@ a {
 }
 
 /* variables */
-:root { --color: red; }
-a { color: var(--color); }
+:root { --color: red; --var-with\.escaped-dot: blue; }
+a { color: var(--color); background: var(--var-with\.escaped-dot);  }
```

Comment:
Minor nitpick, but let's isolate the new declarations in separate rules. This will make it a little easier to identify parsing errors in the future.

```js
:root { --color: red; }
:root { --var-with\.escaped-dot: blue; }
a { color: var(--color); }
a { background: var(--var-with\.escaped-dot); }
```